### PR TITLE
DOCS-2279: Add connection URI reference

### PR DESCRIPTION
**Pull request by andf-viam**
Update `viam data database hostname` reference and usage docs to reflect new returned Connection URI value.
- Converted query and visualize guidance to use connection URI directly, simpler.
	- But retained mention of hostname and database name for visualize guidance, in case some third party tools use hostname / db name for connecting.
- Updated term "connection string" to "connection URI" to match command output.

### CODE REVIEW

**Comment by andf-viam** [comment]
Hi @n0nick -- here's a quick docs update to reflect your work on DATA-2465 (new connection URI returned to user). Let me know if you'd like to see any changes -- thanks!

File: docs/data/query.md [hunk 1]
```diff
@@ -112,48 +112,38 @@ You do not need to perform any additional configuration when [querying data in t
...
-1. Determine the hostname for your organization's MongoDB Atlas Data Federation instance by running the following command with the organization's `org-id` from step 2:
+1. Determine the connection URI for your organization's MongoDB Atlas Data Federation instance by running the following command with the organization's `org-id` from step 2:
 
    ```sh {class="line-numbers linkable-line-numbers"}
    viam data database hostname --org-id=<YOUR-ORGANIZATION-ID>
    ```
 
-   This command returns the `hostname` (including database name) to use to connect to your data store on the Viam organization's MongoDB Atlas instance.
+   This command returns the hostname (including database name) and connection URI to use to connect to your data store on the Viam organization's MongoDB Atlas instance.
+   The connection URI includes everything you need to connect to your organization's MongoDB Atlas Data Federation instance from a valid MQL-compatible client, except your Viam account password.
```
Comment (n0nick) [inline docs/data/query.md:122]: (here and elsewhere) the password required for connection will be the configured DB password (via `viam data database configure`), _not_ the user account's password. Saying "Viam account password" might be confusing?

Comment (andf-viam) [reply docs/data/query.md]: @n0nick Whoops, you're right, thank you -- fixed!

File: docs/data/query.md [hunk 2]
```diff
@@ -119,8 +120,8 @@ You do not need to perform any additional configuration when [querying data in t
    ```
 
    This command returns the hostname (including database name) and connection URI to use to connect to your data store on the Viam organization's MongoDB Atlas instance.
-   The connection URI includes everything you need to connect to your organization's MongoDB Atlas Data Federation instance from a valid MQL-compatible client, except your Viam account password.
-   Replace the placeholder text `YOUR-PASSWORD-HERE` in the connection URI with your Viam account password.
+   The connection URI includes everything you need to connect to your organization's MongoDB Atlas Data Federation instance from a valid MQL-compatible client, except configured database password.
+   Replace the placeholder text `YOUR-PASSWORD-HERE` in the connection URI with the configured database password, as returned from the `viam data database configure` command.
```
Comment (n0nick) [inline docs/data/query.md]: password is not returned, should be "as set by" I think

**Review by n0nick — APPROVED** [review]
LGTM, thanks! 😄  just one minor comment

File: docs/data/query.md [hunk 3]
```diff
@@ -110,50 +110,41 @@ You do not need to perform any additional configuration when [querying data in t
    viam data database configure --org-id=<YOUR-ORGANIZATION-ID> --password=<NEW-DBUSER-PASSWORD>
    ```
 
-   This command configures a new database user for your org for use with data query.
+   This command configures a new database user for your org for use with data query, and sets the password.
+   If you have already run this command before, this command instead updates the password to the new value you set.
 
-1. Determine the hostname for your organization's MongoDB Atlas Data Federation instance by running the following command with the organization's `org-id` from step 2:
+1. Determine the connection URI for your organization's MongoDB Atlas Data Federation instance by running the following command with the organization's `org-id` from step 2:
```
Comment (JessamyT) [inline docs/data/query.md:116]: [opt] Does the user necessarily know what a URI is? They can look it up I suppose

File: docs/data/query.md [hunk 4]
```diff
@@ -110,50 +110,41 @@ You do not need to perform any additional configuration when [querying data in t
...
 
    ```sh {class="line-numbers linkable-line-numbers"}
    viam data database hostname --org-id=<YOUR-ORGANIZATION-ID>
    ```
 
-   This command returns the `hostname` (including database name) to use to connect to your data store on the Viam organization's MongoDB Atlas instance.
+   This command returns the hostname (including database name) and connection URI to use to connect to your data store on the Viam organization's MongoDB Atlas instance.
+   The connection URI includes everything you need to connect to your organization's MongoDB Atlas Data Federation instance from a valid MQL-compatible client, except for your configured database password.
```
Comment (JessamyT) [inline docs/data/query.md:123]: [opt] "everything you need to connect" seems a bit vague? I don't know the topic well enough to provide a good suggestion, so feel free to disregard but maybe rephrase this along the lines of 
```suggestion
   The connection URI includes the identification information (hostname, database name, and username) you need to connect to your organization's MongoDB Atlas Data Federation instance from a valid MQL-compatible client, except for your configured database password.
```
or, since "uniform resource **identifier**" already implies identification, maybe this:
```suggestion
   In addition to the connection URI, to connect to your organization's MongoDB Atlas Data Federation instance from a valid MQL-compatible client you need your configured database password.
```
```suggestion
   To connect to your organization's MongoDB Atlas Data Federation instance from a valid MQL-compatible client, you need to provide both the URI and your configured database password.
```

File: docs/data/query.md [hunk 5]
```diff
@@ -110,50 +110,41 @@ You do not need to perform any additional configuration when [querying data in t
...
+   Replace the placeholder text `YOUR-PASSWORD-HERE` in the connection URI with your configured database password, as set by the `viam data database configure` command.
    You will need this to query your data in the next section.
 
 For more information, see the documentation for the [Viam CLI `database` command](/cli/#data).
 
 ### Query
 
-Once you have synced tabular data to the Viam app, you can directly query that data from an MQL-compatible client, such as [`mongosh`](https://www.mongodb.com/docs/mongodb-shell/) or [Compass](https://www.mongodb.com/docs/compass/current/).
+Once you have synced tabular data to the Viam app, you can directly query that data from an MQL-compatible client, such as the [`mongosh` shell](https://www.mongodb.com/docs/mongodb-shell/), [MongoDB Compass](https://www.mongodb.com/docs/compass/current/), or many third-party tools.
```
Comment (JessamyT) [inline docs/data/query.md:131]: [opt] This is grammatically wonky but my suggestions make it slightly less clear so idk
```suggestion
Once you have synced tabular data to the Viam app, you can directly query that data from an MQL-compatible client, such as the [`mongosh` shell](https://www.mongodb.com/docs/mongodb-shell/), [MongoDB Compass](https://www.mongodb.com/docs/compass/current/), or a third-party tool.
```
```suggestion
Once you have synced tabular data to the Viam app, you can directly query that data from an MQL-compatible client, such as the [`mongosh` shell](https://www.mongodb.com/docs/mongodb-shell/), [MongoDB Compass](https://www.mongodb.com/docs/compass/current/), or one of many third-party tools.
```

File: docs/data/query.md [hunk 6]
```diff
@@ -110,50 +110,41 @@ You do not need to perform any additional configuration when [querying data in t
...
 
-1. Determine the hostname for your organization's MongoDB Atlas Data Federation instance by running the following command with the organization's `org-id` from step 2:
+1. Determine the connection URI for your organization's MongoDB Atlas Data Federation instance by running the following command with the organization's `org-id` from step 2:
 
    ```sh {class="line-numbers linkable-line-numbers"}
    viam data database hostname --org-id=<YOUR-ORGANIZATION-ID>
    ```
 
-   This command returns the `hostname` (including database name) to use to connect to your data store on the Viam organization's MongoDB Atlas instance.
+   This command returns the hostname (including database name) and connection URI to use to connect to your data store on the Viam organization's MongoDB Atlas instance.
```
Comment (JessamyT) [inline docs/data/query.md:122]: Doesn't URI include hostname?

File: docs/data/visualize.md [hunk 7]
```diff
@@ -47,11 +47,28 @@ With data query configured, you can use a variety of popular third-party tools t
 
 When you sync captured data to Viam, that data is stored in the Viam organization’s MongoDB Atlas Data Federation instance.
 
-Most third-party visualization tools require the _connection string_ to that database server, and the _credentials_ to authenticate to that server in order to visualize your data.
+Most third-party visualization tools require the _connection URI_ (also called the connection string) to that database server, and the _credentials_ to authenticate to that server in order to visualize your data.
```
Comment (JessamyT) [inline docs/data/visualize.md:50]: Oooh this section answers some of the questions I had when I first started reviewing this PR. Will users who land on the query page but don't need to visualize need to find this section?

File: docs/cli.md [hunk 8]
```diff
@@ -336,7 +336,7 @@ done
 | ----------- | ----------- | ----------- |
 | `export`      | export data in a specified format to a specified location  | - |
 | `database configure`      | create a new database user for the Viam organization's MongoDB Atlas Data Federation instance, or change the password of an existing user. See [Configure data query](/data/query/#configure-data-query)  | - |
-| `database hostname`      | get the MongoDB Atlas Data Federation instance hostname and database name. See [Configure data query](/data/query/#configure-data-query)  | - |
+| `database hostname`      | get the MongoDB Atlas Data Federation instance hostname and connection URI. See [Configure data query](/data/query/#configure-data-query)  | - |
```
Comment (JessamyT) [inline docs/cli.md:339]: I find myself confused as to why it provides both the hostname and the connection URI if the latter includes the former (per the visualize content)

**Review by JessamyT — COMMENTED** [review]
The visualize page content you add here seems very helpful. I am a very naive reviewer on this topic--I hope my points of confusion are useful :D

File: docs/data/query.md (hunk 4 above)
Comment (andf-viam) [reply docs/data/query.md:123]: 'identifier' in this case only refers to identifying the network resource to connect to, and doesn't infer user identification (even though it also includes the username in the connection string). You can user a URI that doesn't include a UN or PW (if you provide it elsewhere, like with a SQL client that has separate user auth fields from DB connection string field.

I like your first suggestion, which helps to clarify that the connection URI contains everything we need to connect, including the other returned data object `hostname`, which itself is only part of what we'd need to connect. Thanks!

File: docs/data/query.md (hunk 3 above)
Comment (andf-viam) [reply docs/data/query.md:116]: Clarified == connection string here, thanks!

File: docs/data/query.md (hunk 6 above)
Comment (andf-viam) [reply docs/data/query.md:122]: Engineering chose to retain the original output alongside the additional new output. Ostensibly, its so that users used to or expecting the previous output still have access to it, while adding additional data output to benefit those users who seek the new form of this data output.

It is also beneficial for situations where the user does not need the URI (legacy SQL clients that ask for server hostname, database name, username, and password in 4 separate GUI fields. Like [DBeaver](https://dbeaver.com/wp-content/uploads/2022/03/Screen-Shot-2022-03-03-at-11.52.55-AM.png), etc). Saying something like "take the connection URI from the command above and delete a bunch of text we don't need to get at the text we do need" would be more cumbersome for them then simplifying output to neat single data return for us.

File: docs/data/visualize.md (hunk 7 above)
Comment (andf-viam) [reply docs/data/visualize.md:50]: Great callout, this sentence also belongs over in `query`, thank you!

File: docs/cli.md (hunk 8 above)
Comment (andf-viam) [reply docs/cli.md:339]: I gotcha, see comment above.
